source/grid.js

```diff
@@ -153,22 +172,43 @@ this.addToGridFromWorld = function( object, worldCoord ) {
 }
 
 this.removeFromGrid = function( nodeID, gridCoord ) {
-    var objects = this.getObjectsAtCoord( gridCoord );
-    var index = objects.indexOf( nodeID );
-    if ( index !== -1 ) {
-        for ( var i = index; i < objects.length - 1; i++ ) {
-            objects[ i ] = objects[ i + 1 ];
+
+    var node = this.scene.findByID( this.scene, nodeID ); 
+    var bArea = node.boundingAreaSize; 
+
+    for( var x = 0; x < bArea[ 0 ]; x++ ){
+        for( var y = 0; y < bArea[ 1 ]; y++ ){
+            var currTileCoord = [ gridCoord[ 0 ] + x , gridCoord[ 1 ] - y ]; 
+            var objects = this.getObjectsAtCoord( currTileCoord );
+            var index = objects.indexOf( nodeID );
+
+            if ( index !== -1 ) {
+                for ( var i = index; i < objects.length - 1; i++ ) {
+                    objects[ i ] = objects[ i + 1 ];
+                }
+                objects.length--;
+            } else {
+                this.logger.errorx( "removeFromGrid", "Unable to find tile with " +
+                    "ID: " + nodeID );
+                return false;
+            }
```

Comment:
This all worries me. It might be slow, for one, and if it doesn't look at the correct tiles, then we may leave behind IDs causing tiles to be blocked. I'm trying to think of a way that is quicker and safer. Did you test this a lot with multiple turns and verifying that the IDs are properly being removed?